modules/schedule/index.js

```diff
@@ -28,66 +30,75 @@ var lastSchedule = null;
 // }
 
 // m1 - m2 so pass in backwards for positive
-function getDifference(m1,m2) {
-  return moment.duration(moment(m1).diff(moment(m2), "milliseconds"));
-}
+const getDifference = (m1, m2) => (
+  Moment.duration(Moment(m1).diff(Moment(m2), 'milliseconds'))
+);
 
-function getAverageInterval(s) {
-  var numSamples = 5;
-  var next = later.schedule(s).next(numSamples+2); // skip NEXT occurrence
-  var totalSeconds = 0;
+const getAverageInterval = (s) => {
+  const numSamples = 5;
+  let next = later.schedule(s).next(numSamples + 2); // skip NEXT occurrence
+  let totalSeconds = 0;
 
-  next.forEach(function(e,i,d) {
-    e = moment(e);
-  });
-  
-  for(i=1; i<=numSamples; i++)
-  {
-    totalSeconds += parseInt(getDifference(next[i+1], next[i]).format("s"));
+  next = next.map(sched => new Moment(sched));
+
+  for (let i = 1; i <= numSamples; i++) {
+    totalSeconds += parseInt(getDifference(next[i + 1], next[i]).format('s'), 10);
   }
 
-  return totalSeconds/numSamples;
-}
+  return totalSeconds / numSamples;
+};
 
-function writeSchedule(data) {
-  bot.writeDataFile("later.json", 
-      JSON.stringify(schedules), function(err) {
-        if(err) console.log("Error writing command file: " + err);
+const writeSchedule = (data) => {
+  bot.writeDataFile('later.json', JSON.stringify(data), (err) => {
+    if (err) {
+      log.debug(`Error writing command file: ${err}`);
+    }
   });
-}
+};
+
+const sortSchedules = scheds => (
+  Object.values(scheds).sort((a, b) => {
+    if (a.created > b.created) {
+      return 1;
+    }
+    if (a.created < b.created) {
+      return -1;
+    }
+
+    return 0;
+  })
+);
 
 function newSchedule(data) {
-  var s;
+  let s;
 
   // reminders use cron
-  if(data.calls == -1)
-  {
+  if (data.calls === -1) {
     try {
       s = later.parse.text(data.schedule);
     } catch (ex) {
       return ex;
     }
 
-    if(typeof(s) == "number" || s.error >= 0 )
-    {
-      var m = "Provided schedule query doesn't parse:\n";
-      var offset = 0;
+    if (typeof(s) === 'number' || s.error >= 0) {
+      let m = 'Provided schedule query doesn\'t parse:\n';
```

Comment:
eslint with airbnb throws a fit if you use backticks without any interpolation.

Reply:
Because of https://github.com/airbnb/javascript/blob/eslint-config-airbnb-v16.1.0/packages/eslint-config-airbnb-base/rules/style.js#L391-L392, I think that you can actually use double-quotes here and still be on airbnb's good side.